components/raftstore/src/store/fsm/peer.rs

```diff
@@ -688,53 +689,57 @@ where
                 }
             }
         }
+        self.on_loop_finished();
+    }
+
+    #[inline]
+    #[allow(unused_mut, clippy::redundant_closure_call)]
```

Comment:
Why `unused_mut`?

Reply:
The `mut should_propose` is only used in failpoint.

Reply:
You can get around the problem by putting failpoint to or chain.
```
let should_propose = self.ctx.sync_write_worker.is_some()
            || (ready_concurrency != 0
                && self.fsm.peer.unpersisted_ready_len() < ready_concurrency
                && (|| {
                       fail_point!(
                "force_delay_propose_batch_raft_command",
                |_| false
            );
            true
        })();
```

Reply:
Seems more difficult to read. So many brackets.

Reply:
Or
```
let delay_by_fp = (|| {
    fail_point!(
            "force_delay_propose_batch_raft_command",
           self.ctx.sync_write_worker.is_none()
            |_| true
    );
    false
})();
should_propose &= !delay_by_fp;
```